src/vsg/utils/GraphicsPipelineConfigurator.cpp

```diff
@@ -496,18 +551,28 @@ void GraphicsPipelineConfigurator::copyTo(ref_ptr<StateGroup> stateGroup, ref_pt
         {
             if (auto ds = descriptorConfigurator->descriptorSets[set])
             {
-                if (sharedObjects)
+                if (visit<DescriptorSetContainsNull>(ds).containsNull)
                 {
-                    sharedObjects->share(ds);
+                    if (visit<DescriptorSetContainsData>(ds).containsData)
```

Comment:
Is this checked for a case where one DescriptorBuffer/DescriptorImage is empty so triggers the first condition, but there is another DescriptorBuffer/DescriptorImage on the same DesciptorSet that isn't empty?

Is this not a condition where defaults is the appropriate to fill in the missing entries rather than have valid entries just ignored?

Or is this really a case of an invalid application usage?

Reply:
> Is this checked for a case where one DescriptorBuffer/DescriptorImage is empty so triggers the first condition, but there is another DescriptorBuffer/DescriptorImage on the same DesciptorSet that isn't empty?

Yes.
> 
> Is this not a condition where defaults is the appropriate to fill in the missing entries rather than have valid entries just ignored?

In this case there aren't any defaults, or they would already be in the list of descriptors. In other words, the only way a descriptor can have an empty buffer or image info list is if the shader set does not provide a default, or the user assigns such an empty descriptor.

> 
> Or is this really a case of an invalid application usage?

The comment alludes to the possibility of null as a valid value if certain features and extensions are enabled, but I haven't yet wrapped my around the functionality that's actually provided.

The motivation for this whole ceremony is that the descriptor set layout for the pipeline needs to be constructed, but the user may leave some descriptor sets empty. I submitted ShaderSet::createPipelineLayout() in another PR which builds the layout directly from the list of defines. I might be useful here.

Reply:
I wonder if a debug method of some kind might be helpful to give users better insight into how the ShaderSet/GraphicsPipelineConfigurator usage is incomplete.  I think it's a bit premature for me to start looking at this thought, first I need to get the extra controls we need for the vsgCs usage approach working.